routers/routes/routes.go

```diff
@@ -485,6 +485,31 @@ func RegisterRoutes(m *macaron.Macaron) {
 				return
 			}
 
+			repository, unitType, err := attach.LinkedRepository()
```

Comment:
I think we should move this closure function to a real function since it's not small yet.